thunderpb/dependency.proto

```diff
@@ -0,0 +1,41 @@
+syntax = "proto3";
+
+package thunderpb;
+option go_package = "github.com/samsarahq/thunder/thunderpb";
+
+import "github.com/gogo/protobuf/gogoproto/gogo.proto";
+import "google/protobuf/timestamp.proto";
+
+enum FieldKind {
+  Unknown = 0;
+  Null = 1;
+  Bool = 2;
+  Int = 3;
+  Uint = 4;
+  String = 5;
+  Bytes = 6;
+  Float64 = 7;
+  Time = 8;
+}
+
+message Field {
```

Comment:
ahh, think I found what's causing a lot of cloudserver OOMs, I think this field is quite large and taking up much more memory than usual whenever we create a dependency.

Have we thought of using the proto oneof annotation here?  Not sure if it actually reduces the allocation in code, just a thought.

Reply:
here's one of our heap profiles: https://s3.internal.samsara.com/s3/samsara-prod-profile/2018-10-02/cloudserver/1438-prod-23996c8a-f0f5-47be-960a-d96d639a2f26-2wHXqesa-heap.pprof.svg